### PR TITLE
Adding a new execution module to manipulate JIRA tickets

### PR DESCRIPTION
**Pull request by mirceaulinic**
### What does this PR do?

Adding a new execution module to manipulate JIRA tickets - very useful to dynamically create tickets as in response to various events (e.g., create ticket on interface down, or on server reboot etc.).

### CODE REVIEW

**Comment by damon-atkins** [comment]
Please rename jira_mod to atlassian_jira

File: salt/modules/jira_mod.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+Configuration example:
+
+.. code-block:: yaml
+
+  jira:
+    server: https://jira.atlassian.org
+    username: salt
+    password: pass
+'''
+from __future__ import absolute_import
```
Comment (rallytime) [inline salt/modules/jira_mod.py]: `print_function` and `unicode_literals` need to be added here for imports

**Review by rallytime — CHANGES_REQUESTED** [review]
One small thing to fix, otherwise this looks good to me. The name of the module is fine with me.

File: salt/modules/jira_mod.py (hunk 1 above)
Comment (mirceaulinic) [reply salt/modules/jira_mod.py]: Fixed @rallytime

**Review by rallytime — APPROVED** [review]

**Comment by damon-atkins** [comment]
Reason for the name change e.g. atlassian_jira is so other atlassian modules can have the name atlassian_* pattern.    And also mod not need in the name of a module as its in the module directory.

**Comment by cachedout** [comment]
I don't agree with the proposed naming. We don't, for example, name the `apache` module `apache_foundation_apache` simply because the Apache Foundation produces other software. Jira is well-known on its own and I just don't see how including the vendor name adds any clarity or organizational value.

**Pull request merged**